pkg/controller/managedresources/merger.go

```diff
@@ -39,6 +39,12 @@ func merge(desired, current *unstructured.Unstructured, forceOverwriteLabels boo
 	newObject.SetResourceVersion(oldObject.GetResourceVersion())
 	newObject.SetFinalizers(oldObject.GetFinalizers())
 
+	// retain metadata information to avoid unnecessary update calls
+	newObject.SetCreationTimestamp(oldObject.GetCreationTimestamp())
+	newObject.SetSelfLink(oldObject.GetSelfLink())
+	newObject.SetUID(oldObject.GetUID())
+	newObject.SetGeneration(oldObject.GetGeneration())
```

Comment:
There are also other fields in the `metadata`, e.g. `clusterName`, `managedFields`, etc. What about them? Wouldn't it be safer to copy the whole metadata and then simply overwrite the fields that we want to overwrite? Though, not sure if that's possible?

Reply:
Valid point, I changed it to copying the whole `metadata` stanza and tested it/played around with it a wile and it seems to work fine :)